crates/driver/src/infra/api/routes/settle/mod.rs

```diff
@@ -8,41 +8,93 @@ use {
+            if state.eth().current_block().borrow().number >= req.submission_deadline_latest_block {
+                if let Err(err) =
+                    response_sender.send(Err(competition::Error::QueueAwaitingDeadlineExceeded))
+                {
+                    tracing::error!(
+                        ?err,
+                        "settle deadline exceeded. unable to return a response"
+                    );
+                }
+                continue;
+            }
+
+            let result = async move {
+                observe::settling();
+                let result = state
+                    .competition()
+                    .settle(
+                        req.auction_id,
+                        req.solution_id,
+                        req.submission_deadline_latest_block,
+                    )
+                    .await;
+                observe::settled(state.solver().name(), &result);
+                result.map(|_| ()).map_err(Into::into)
+            }
+            .instrument(tracing::info_span!("/settle", solver, auction_id))
+            .await;
+
+            if let Err(err) = response_sender.send(result) {
+                tracing::error!(?err, "Failed to send /settle response");
+            }
+        }
+    });
+
+    sender
+}
+
 async fn route(
     state: axum::extract::State<State>,
     req: axum::Json<dto::Solution>,
 ) -> Result<(), (hyper::StatusCode, axum::Json<Error>)> {
-    let auction_id = req.auction_id;
-    let solver = state.solver().name().to_string();
-
-    let handle_request = async move {
-        observe::settling();
-        let result = state
-            .competition()
-            .settle(
-                req.auction_id,
-                req.solution_id,
-                req.submission_deadline_latest_block,
-            )
-            .await;
-        observe::settled(state.solver().name(), &result);
-        result.map(|_| ()).map_err(Into::into)
-    }
-    .instrument(tracing::info_span!("/settle", solver, auction_id));
-
-    // Handle `/settle` call in a background task to ensure that we correctly
-    // submit the settlement (or cancellation) on-chain even if the server
-    // aborts the endpoint handler code.
-    // This can happen due do connection issues or when the autopilot aborts
-    // the `/settle` call when we reach the submission deadline.
-    Ok(tokio::task::spawn(handle_request)
-        .await
-        .unwrap_or_else(|_| Err(competition::Error::SubmissionError))?)
+    let sender = state.settle_queue_sender();
+    let (response_tx, response_rx) = oneshot::channel();
+
+    let queued_request = QueuedSettleRequest {
+        state: state.0.clone(),
+        req: req.0,
+        response_sender: response_tx,
+    };
+
+    sender.send(queued_request).await.map_err(|err| {
+        tracing::error!(?err, "Failed to enqueue /settle request");
+        competition::Error::UnableToEnqueue
+    })?;
+
+    Ok(response_rx.await.map_err(|err| {
```

Comment:
No, since the mempool internally checks for the submission deadline and the task will be dropped once it is reached:
https://github.com/cowprotocol/services/blob/bcd024ea977336e7e178ee4a9890e509cbf963c3/crates/driver/src/domain/mempools.rs#L124-L133